examples/pytorch/speech-recognition/run_speech_recognition_seq2seq.py

```diff
@@ -405,7 +412,7 @@ def prepare_dataset(batch):
         inputs = feature_extractor(sample["array"], sampling_rate=sample["sampling_rate"])
         # process audio length
         batch[model_input_name] = inputs.get(model_input_name)[0]
-        batch["input_length"] = len(inputs.get(model_input_name)[0])
+        batch["input_length"] = len(sample["array"])
```

Comment:
For Whisper, the `inputs_features` are log-mel filter bank coeffs of shape `(80, 3000)` -> we can't take the length of this tensor to get the input length (it would always be 80).

We resample the audio inputs from their original sampling rate to the feature extractor sampling rate:
https://github.com/huggingface/transformers/blob/0b7b07ef033ec3c9733a09ea06903a545f5e4c9b/examples/pytorch/speech-recognition/run_speech_recognition_seq2seq.py#L364

We also compute the `max_input_length` based on the feature extractor sampling rate: https://github.com/huggingface/transformers/blob/0b7b07ef033ec3c9733a09ea06903a545f5e4c9b/examples/pytorch/speech-recognition/run_speech_recognition_seq2seq.py#L372

=> the audio array and `max_input_length` are aligned based on sampling rate, so we can take the input length as the length of the audio array and compare with the `max_input_length`

cc @patrickvonplaten